coderd/users.go

```diff
@@ -519,13 +527,15 @@ func (api *API) putUserRoles(rw http.ResponseWriter, r *http.Request) {
 	added, removed := rbac.ChangeRoleSet(roles.Roles, impliedTypes)
 	for _, roleName := range added {
 		// Assigning a role requires the create permission.
-		if !api.Authorize(rw, r, rbac.ActionCreate, rbac.ResourceRoleAssignment.WithID(roleName)) {
+		if !api.Authorize(r, rbac.ActionCreate, rbac.ResourceRoleAssignment.WithID(roleName)) {
+			httpapi.Forbidden(rw)
 			return
 		}
 	}
 	for _, roleName := range removed {
 		// Removing a role requires the delete permission.
-		if !api.Authorize(rw, r, rbac.ActionDelete, rbac.ResourceRoleAssignment.WithID(roleName)) {
+		if !api.Authorize(r, rbac.ActionDelete, rbac.ResourceRoleAssignment.WithID(roleName)) {
+			httpapi.Forbidden(rw)
```

Comment:
leaks existence of user

Reply:
Oh this is an annoying one. If you don't have permissions to grant/delete a role, we return a 404 then?

Reply:
I made it so if you cannot read the user, we return a 404